soda_machine.rb

```diff
@@ -7,12 +9,38 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+
+    count = 0
+
+    @sodas.each { count += 1 }
+
+    return count
+
   end
 
   def find_soda(soda_brand)
+
+    @sodas.each_with_index do |soda, idx|
+      if soda.brand == soda_brand
+        return @sodas[idx]
+      end
+    end
+
+    return nil
   end
 
   def sell(soda_brand)
+
+    @sodas.each_with_index do |soda, idx|
+      if soda.brand == soda_brand
+        temp = @sodas[idx]
+        @sodas.delete_at(idx)
+        @cash += soda.price
+        return temp
+      end
+    end
+
+    return nil
   end
```

Comment:
You can reuse the find_soda method here instead of defining the whole login again.

```
def sell ( soda_brand )
   soda = find_soda(soda_brand)
   if soda
     @cash += soda.price
     @sodas.delete(soda)
   else
     nil
   end
end
```